ledger/participant-integration-api/src/test/lib/scala/platform/store/backend/StorageBackendTestValues.scala

```diff
@@ -205,25 +205,61 @@ private[backend] object StorageBackendTestValues {
     )
   }
 
+  sealed trait Deduplication extends Product with Serializable {
+    import Deduplication.{Offset => DedupOffset, _}
+
+    def startOffset: Option[String] = this match {
+      case DedupOffset(offset) => Some(offset)
+      case _ => None
+    }
+    def duration: Option[SecondsAndNanos] = this match {
+      case Span(span) => Some(span)
+      case _ => None
+    }
+    def startInstant: Option[Instant] = this match {
+      case Start(instant) => Some(instant)
+      case _ => None
+    }
+  }
+  object Deduplication {
+    final case class SecondsAndNanos(seconds: Long, nanos: Int) {
+      def allComponentsAreNonNegative: Boolean = seconds >= 0 && nanos >= 0
+    }
+
+    final case class Offset(offset: String) extends Deduplication
+    final case class Span(span: SecondsAndNanos) extends Deduplication {
+      require(
+        span.allComponentsAreNonNegative,
+        s"All the deduplication window components must not be negative: $span",
+      )
+    }
+    final case class Start(instant: Instant) extends Deduplication
+  }
+
   def dtoCompletion(
```

Comment:
> wouldn't it will be simpler just to extend dtoCompletion with the default null-s, and override the needed ones in the test

Yes and necessary to test for unexpected data. Done so.